corehq/apps/users/views/utils.py

```diff
@@ -113,26 +115,58 @@ def log_commcare_user_locations_changes(request, user, old_location_id, old_assi
         )
 
 
-def get_locations_with_cases(domain, location_ids):
-    """
-    Takes a list of location IDs and returns how many cases are assigned to each location
-    :param domain: The domain to search in
-    :param location_ids: A list of location IDs to get case counts on
-    :returns A dict with location names as the key, and the number of cases assigned to it as the value
-    """
+def _get_location_ids_with_single_user(domain, location_ids, user_id):
+    # Remove the user's ID, as we want to see which other CommCare users share any of the given location_ids
+    user_query = (
+        UserES()
+        .domain(domain)
+        .mobile_users()
+        .location(location_ids)
+    )
+    user_query.filter(filters.NOT(filters.doc_id(user_id)))
+    other_commcare_user_ids = user_query.get_ids()
+
+    case_query = (
+        CaseES()
+        .domain(domain)
+        .owner(location_ids)
+        .user(other_commcare_user_ids)
+        .aggregation(TermsAggregation('by_location', 'owner_id')
+        .size(0))
+    ).run()
+    return list(set(location_ids) - set(case_query.aggregations.by_location.keys))
+
+
+def _get_location_case_counts_with_single_user(domain, location_ids):
     query = (CaseES()
             .domain(domain)
             .owner(location_ids)
             .aggregation(TermsAggregation('by_location', 'owner_id')
-            .size(0)))
-    counts = query.run().aggregations.by_location.counts_by_bucket()
-    locations_with_cases = dict()
-    if counts:
-        locations = SQLLocation.objects.filter(location_id__in=location_ids)
+            .size(0))
+    ).run()
+    locations = SQLLocation.objects.filter(location_id__in=location_ids)
+    counts = query.aggregations.by_location.counts_by_bucket()
+    return locations, counts
+
+
+def get_locations_with_single_user(domain, location_ids, user_id):
+    """
+    Takes a list of location IDs and returns all the given location IDs
+    where the user ID is the only one that has cases there.
+    :param domain: The domain to search in.
+    :param location_ids: A list of location IDs to get case counts on.
+    :param user_id: The user ID to check given location_ids with as the only assigned user with cases.
+    :returns A dict with location names as the key, and the number of cases assigned to them as the value.
```

Comment:
Location names aren't guaranteed to be unique except among sibling locations, so this could in theory mess things up.

Reply:
That is a good point. In that scenario, the alert would essentially only show the case count for the last location that has the identical name. Do you think that adding a parent location name prefix such as "South Africa/Cape Town" could work in situations such as these? Although this could also be tricky if, in the unlikely scenario, both identical location names also share the same parent location name.

Reply:
You could return a list of tuples rather than a dictionary, or use `(loc.pk, loc.name)` as the key and only display the name, or use `loc.get_path_display()`, which does `South Africa/Cape Town` for the whole path to the location.

Reply:
That could definitely work, thanks for the suggestion. I'll take a look at implementing something like that. :+1: